x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_creation/engine_creation_logic.ts

```diff
@@ -112,25 +152,97 @@ export const EngineCreationLogic = kea<MakeLogicType<EngineCreationValues, Engin
         setEngineType: (_, { engineType }) => engineType,
       },
     ],
+    currentEngineCreationStep: [
+      EngineCreationSteps.SelectStep,
+      {
+        setCreationStep: (_, currentEngineCreationStep) => currentEngineCreationStep,
+      },
+    ],
   },
   selectors: ({ selectors }) => ({
     name: [() => [selectors.rawName], (rawName) => formatApiName(rawName)],
+    aliasName: [() => [selectors.aliasRawName], (aliasRawName) => formatApiName(aliasRawName)],
     indicesFormatted: [
       () => [selectors.indices, selectors.selectedIndex],
       (indices: ElasticsearchIndex[], selectedIndexName) =>
         formatIndicesToSelectable(indices, selectedIndexName),
     ],
     isSubmitDisabled: [
-      () => [selectors.name, selectors.engineType, selectors.selectedIndex],
-      (name: string, engineType: EngineType, selectedIndex: string) =>
-        (name.length === 0 && engineType === 'appSearch') ||
-        ((name.length === 0 || selectedIndex.length === 0) && engineType === 'elasticsearch'),
+      () => [
+        selectors.name,
+        selectors.engineType,
+        selectors.selectedIndex,
+        selectors.aliasName,
+        selectors.showAliasNameErrorMessages,
+      ],
+      (
+        name: string,
+        engineType: EngineType,
+        selectedIndex: string,
+        aliasName: string,
+        showAliasNameErrorMessages: boolean
+      ) => {
+        if (name.length === 0 || showAliasNameErrorMessages) {
+          return true;
+        }
+
+        if (engineType === 'elasticsearch') {
+          if (selectedIndex.length === 0) {
+            return true;
+          }
+
+          if (aliasName.length === 0) {
+            return !selectedIndex.startsWith('search-');
+          } else {
+            return !aliasName.startsWith('search-');
+          }
+        }
+
+        return false;
+      },
+    ],
+    isAliasRequired: [
+      () => [selectors.selectedIndex],
+      (selectedIndex: string) => selectedIndex.length > 0 && !selectedIndex?.startsWith('search-'),
+    ],
+    selectedIndexFormatted: [
+      () => [selectors.selectedIndex, selectors.indicesFormatted],
+      (selectedIndex: string, indicesFormatted: SearchIndexSelectableOption[]) => {
+        return indicesFormatted.find((el) => el.label === selectedIndex);
+      },
+    ],
+    aliasNameErrorMessage: [
+      () => [selectors.aliasName, selectors.indices],
+      (aliasName: string, indices: ElasticsearchIndex[]) => {
+        const existingAlias = indices.find((el) => el.name === aliasName);
+        if (existingAlias) {
+          return i18n.translate(
+            'xpack.enterpriseSearch.appSearch.engineCreation.configureForm.backButton.label',
```

Comment:
`backButton.label` doesn't feel like the right id for this string

Reply:
Great find.